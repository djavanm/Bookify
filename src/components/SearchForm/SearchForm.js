import React, { Component } from 'react';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      searchInput: ''
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: [e.target.value]});
  }

  render() {
    const { searchInput } = this.state;
    
    return (
      <form>
        <input 
        type="text" 
        name="searchInput"
        value={searchInput}
        onChange={this.handleChange}/>
        <button>Search</button>
      </form>
    )
  }
}

export default SearchForm;