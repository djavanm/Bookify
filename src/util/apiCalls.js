export const fetchOnLoad = () => {
  return fetch('https://itunes.apple.com/search?term=audiobooks&limit=10')
    .then(response => {
      if(!response.ok) {
        throw Error('ERROR')
      }
      return response.json()
    })
    .catch(error => Error(error.message))
}

export const getBooks = (query) => {
  return fetch(`https://itunes.apple.com/search?media=audiobook&term=${query}`)
    .then(response => {
      if(!response.ok) {
        throw Error('ERROR')
      }
      return response.json()
    })
    .catch(error => Error(error.message))
}
