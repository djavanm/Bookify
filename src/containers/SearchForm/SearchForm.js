import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setBooks, showStart } from '../../actions';
import { getBooks } from '../../util/apiCalls'

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      searchInput: ''
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: [e.target.value]});
  }

  handleSearch = (e) => {
    const { searchInput } = this.state;
    const { setBooks, showStart, length } = this.props;
    e.preventDefault();
    getBooks(searchInput)
      .then(books => setBooks(books))
      .catch(error => console.log(error))
    this.setState({
      searchInput: ''
    })
    showStart({start: 0, end: 10, length: length})
  }

  render() {
    const { searchInput } = this.state;

    return (
      <form>
        <input
        type="text"
        name="searchInput"
        value={searchInput}
        onChange={this.handleChange}/>
        <button onClick={this.handleSearch}>Search</button>
      </form>
    )
  }
}

export const mapStateToProps = state => ({
  length: state.currentBooks.length
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({ setBooks, showStart }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(SearchForm);
