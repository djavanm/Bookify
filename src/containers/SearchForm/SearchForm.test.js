import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm } from './SearchForm';
import { getBooks } from '../../util/apiCalls';

jest.mock('../../util/apiCalls');

getBooks.mockImplementation(() => {
  return Promise.resolve({
    id: 1,
    book_id: 1,
    author_name: 'Djavan',
    book_name: 'How to do apiCalls, version 1',
    artwork_url: 'image string that goes here',
    release_date: '9999',
    description: 'The description on this is so descriptive',
    primary_genre_name: 'Loud'
  })
});

describe('SearchForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchForm />)
  });

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the state when handle change is called', () => {
    const mockEvent = {
      target: {
        name: 'searchInput',
        value: 'harry potter'
      }
    };
    const expectedSearch = 'harry potter';
     wrapper.instance().handleChange(mockEvent);
     expect(wrapper.state('searchInput')).toEqual(expectedSearch);
  });

  it('should call the getBooks fetch when the search form is submitted', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    }
    wrapper.instance().handleSearch(mockEvent);
    expect(getBooks).toHaveBeenCalled();
  });
});
