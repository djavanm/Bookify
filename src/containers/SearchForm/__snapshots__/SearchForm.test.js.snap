// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with the data passed through 1`] = `
<form
  className="search-form"
>
  <input
    className="search-input"
    name="searchInput"
    onChange={[Function]}
    placeholder="Enter an Audiobook or Author"
    type="text"
    value=""
  />
  <button
    className="search-btn"
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
